packages/core/src/validate/run.ts

```diff
@@ -241,13 +243,35 @@ function* getConstructorErrors(contractDef: ContractDefinition, decodeSrc: SrcDe
   }
 }
 
+function initCaches() {
```

Comment:
I would simplify this to just create a single cache, perhaps renaming to `initOpcodeCache`, and `Cache` to `OpcodeCache` to be a little more specific.